### PR TITLE
Add API healthcheck test

### PR DESCRIPTION
**Pull request by stuartleeks**
- Add `make api-healthcheck`
- incorporate healthcheck in CI after TRE Core deployment

Following #1573, add a healthcheck to the build to catch deployments with unhealthy API earlier in the build

### CODE REVIEW

**Comment by stuartleeks** [comment]
Successful run with `deploy_tre_branch.yml`: https://github.com/microsoft/AzureTRE/runs/5659256061?check_suite_focus=true#step:3:1518

**Comment by marrobi** [comment]
Do we need to wait for things to get to a healthy stage? Or is expected all should be healthy when the deployment finished? @anatbal is just about to add the resource processor health to the health endpoint, and cloud init takes a while to finish, but VMSS deployment will have completed.

File: .github/workflows/deploy_tre_reusable.yml [hunk 1]
```diff
@@ -218,7 +218,7 @@ jobs:
       - name: Deploy TRE Core
         uses: ./.github/actions/devcontainer_run_command
         with:
-          COMMAND: "TF_VAR_ci_git_ref=${{ inputs.ciGitRef }} TF_LOG=${{ secrets.TF_LOG }} make deploy-core"
+          COMMAND: "TF_VAR_ci_git_ref=${{ inputs.ciGitRef }} TF_LOG=${{ secrets.TF_LOG }} make deploy-core && make api-healthcheck"
```
Comment (tamirkamara) [inline .github/workflows/deploy_tre_reusable.yml:221]: What about splitting the extra make command into a separate step?

**Comment by stuartleeks** [comment]
At the point where the TRE Core deployment step completes, I think that the resources should be deployed and ready. Otherwise, the project's CI process needs to include a wait, anyone else implementing their own CI to deploy AzureTRE also needs to add that step, and so do people creating a dev environment.

File: .github/workflows/deploy_tre_reusable.yml (hunk 1 above)
Comment (stuartleeks) [reply .github/workflows/deploy_tre_reusable.yml:221]: Yeah, I was on the fence about whether to split or not. Happy to split it out!

Comment (stuartleeks) [reply .github/workflows/deploy_tre_reusable.yml]: Done

**Review by tamirkamara — APPROVED** [review]

**Comment by stuartleeks** [comment]
/test-force-approve

Skipping pr-bot tests as it won't load new workflow. Have tested via `deploy_tre_branch` as per previous comment.

**Pull request merged**